src/services/impl/FileTextWriter.hh

```diff
@@ -17,7 +17,12 @@ class FileTextWriter : public ra::services::TextWriter
 {
 public:
     explicit FileTextWriter(const std::wstring& sFilename) noexcept
-        : m_oStream(sFilename, std::ios::binary)
+        : FileTextWriter(sFilename, std::ios::out)
+    {
+    }
+
+    explicit FileTextWriter(const std::wstring& sFilename, std::ios_base::openmode nMode) noexcept
+        : m_oStream(sFilename, std::ios::binary | nMode)
```

Comment:
Not too sure how to adjust it but all the ostream classes have std::ios::out on by default and gets added to any flags input by the user.
Could be something like this:
```
...
explicit FileTextWriter(const std::wstring& sFilename, std::ios_base::openmode nMode = std::ios_base::binary) noexcept
...
```
Then you could probably get rid of the constructor above and leave the rest of this one the same. 

One other suggestion could be to subclass this class, friend the subclass, and specialize it for binary writing. You could call it `BinaryFileTextWriter` to make it clear it's for binary ops and not for formatted text. It seems kind of weird that this class has one constructor for formatted text and one for binary text.

From what I can see, it seems much more appropriate to make `std::ios::binary` the default open_mode because of `m_oStream.write` everywhere. Mixing a formatted stream and unformatted stream is a bad idea.